### ISSUE TITLE
Part of ghost blog static files are not hosted when packaged

### ISSUE THREAD

**Issue by karaxuna** [jxcore/jxcore#524]
When packaging ghost blog, visitor's page works fine, but on `/ghost` path (admin page) some of the files don't work. See [repository](https://github.com/karaxuna/jxcore-tutorial-ghost-packaged).
![error](https://cloud.githubusercontent.com/assets/2152815/9713104/5611f8f8-5560-11e5-8aeb-735506e0afd3.png)

**Comment by ktrzeciaknubisa** [comment]
I saw some issues with SpiderMonkey: https://github.com/jxcore/jxcore/issues/519#issuecomment-138606792, however V8 was working though. Can you confirm?

**Comment by ktrzeciaknubisa** [comment]
Aside note here: the repository mentioned above shows how to run ghost.jx as fully extractable package. Although it is not necessary to extract all contents for this project, this case reminds, that we shouldn't use `jxcore.utils.cmdSync()` to run the package after being extracted, because e.g. all the console output gets displayed only after extracted process ends, and with ghost (or other http server apps) it is not meant to end soon... I'll look into `cmdSync()` replacement.

The original issue still stays open.

@karaxuna While testing this repo, can you please use 0.3.0.6? Are you able to build it?

**Comment by karaxuna** [comment]
@ktrzeciaknubisa Is it possible to extract only part of package?

**Comment by ktrzeciaknubisa** [comment]
Yes. This is new in 0.3.0.6. [Partial extraction](http://jxcore.com/docs/jxcore-feature-packaging-code-protection.html#partial-extraction), but I'm not sure yet what more (except _*.node_ files) should be extracted for ghost to work. Maybe some db or config folders, etc. This needs to be checked. 

However you may also continue with full extraction. The above commit 06a219591280c9a613b35805dcd2f52046aed983 improves that part. Also probably gets rid of missing files problem.

**Comment by ktrzeciaknubisa** [comment]
BTW. the above applies to V8, because SM still have some problems with ghost.

**Comment by karaxuna** [comment]
@ktrzeciaknubisa partial extraction is best solution. Only `/content` folder is dynamic in ghost, so only it shall be extracted. Maybe config file too. Thanks, I'll get back to this as soon as I can.

**Comment by ktrzeciaknubisa** [comment]
While I didn't see this issue with V8 on full extractable package, I started too see it with partial extraction, unless I'll mark *node_modules/ghost/core/built* also as extractable:

```bash
$ jx package index.js "ghost" --extract-what "*.node,./content,core/built" --extract-overwrite --extract-app-root
```

I'm not sure yet why assets from that folder get timeouted on browser request...

**Comment by ktrzeciaknubisa** [comment]
I've got something weird. Those few files fall under a magic 1024 bytes limit, while other files doesn't. Once I cut their contents into 1023 they are served. I need to dig more.

**Comment by ktrzeciaknubisa** [comment]
Ok, after a long fight, I've found a problem. It is related to `fs.createReadStream()` for packages - especially `stream.pipe()`. I posted my proposal to PR https://github.com/jxcore/jxcore/pull/548.

This passes all our tests too.

**Comment by ktrzeciaknubisa** [comment]
This issue is solved with 505e33f8eec8b9251c88b5676f2bbc88913bf399 .
